independent-projects/arc/runtime/src/main/java/io/quarkus/arc/InvocationContextImpl.java

```diff
@@ -82,9 +87,9 @@ public static InvocationContextImpl aroundConstruct(Constructor<?> constructor,
 
     private final Constructor<?> constructor;
 
-    private Object[] args;
+    private final List<Object> args;
 
-    private int position;
+    private final AtomicInteger position;
```

Comment:
Can you change this to a volatile integer with an AtomicIntegerUpdater instead? This object can be created a lot so it is better to save the allocation.

Reply:
TBH I don't think you even need the updater, if you have multiple threads calling next() concurrently then you are into undefined territory anyway. Just making the field volatile should be enough to make sure the state is visible to a new thread.